torchacc/core/dynamic.py

```diff
@@ -0,0 +1,38 @@
+from typing import List, Union
+
+import torch
+import torch_xla
+from torch._dispatch.python import enable_python_dispatcher
+
+import torchacc as ta
+
+
+def mark_dynamic(x: torch.Tensor, dims: Union[List[int], int],
+                 bounds: Union[List[int], int]):
+    """Mark a tensor as having dynamic dims and set corresponding upper bounds for the dims.
+    Args:
+        x (torch.Tensor): input tensor
+        dims (Union[List[int], int]): the dim marked as dynamic, can be a single dim or
+            multiple dims
+        bounds (Union[List[int], int]): upper bounds corresponding to the dynamic dims
+    """
+    if ta.get_global_context().python_dispatcher is None:
+        # Some symint operators require this in order for the operator dispatch
+        # to work properly
+        ta.get_global_context().python_dispatcher = enable_python_dispatcher()
+    if isinstance(dims, int):
+        dims = [dims]
+        bounds = [bounds]
+    assert isinstance(dims, list), "dims should be of int or list type"
```

Comment:
Replace "dims should be of int or list type" with "dims should be of type int or list of int"

Reply:
done